### PR TITLE
Add missing `require` statements for the API

### PR DESCRIPTION
**Pull request by Maroo-b**
The purpose of this PR is to add missing `require` statements in examples, it addresses this [issue](https://github.com/crystal-lang/crystal/issues/7487)

I went through the current docs (0.27.2) and check all the provided examples.

### Issues
Iterator example: 
[link](https://github.com/crystal-lang/crystal/blob/master/src/iterator.cr#L180)
```
array_of_iters = [[1], [2, 3], [4, 5, 6]]
iter = Iterator(Int32).chain array_of_iters
ter.next # => 1
iter.next # => 2
iter.next # => 3
iter.next # => 4
```

returns this error:
```
instantiating 'Iterator::ChainsAll(Array(Int32), Int32)#next()'
src/iterator.cr:207: undefined method 'next' for Array(Int32)
```

### CODE REVIEW

**Comment by Sija** [comment]
IMHO `require` statements in docs should be added only at the module level. Repeating those for **every** example seems repetitive and unnecessary.

File: src/big/big_int.cr [hunk 1]
```diff
@@ -350,6 +352,7 @@ struct BigInt < Int
   # Returns a string representation of self.
   #
   # ```
+  # require "big"
```
Comment (bcardiff) [inline src/big/big_int.cr]:
```suggestion
  # require "big"
  #  
```

File: src/big/big_rational.cr [hunk 2]
```diff
@@ -130,6 +130,7 @@ struct BigRational < Number
   # Divides the rational by (2 ** *other*)
   #
   # ```
+  # require "big"
```
Comment (bcardiff) [inline src/big/big_rational.cr]:
```suggestion
  # require "big"
  #  
```

File: src/big/big_rational.cr [hunk 3]
```diff
@@ -139,6 +140,7 @@ struct BigRational < Number
   # Multiplies the rational by (2 ** *other*)
   #
   # ```
+  # require "big"
```
Comment (bcardiff) [inline src/big/big_rational.cr]:
```suggestion
  # require "big"
  #  
```

**Comment by Maroo-b** [comment]
@bcardiff thanks for the quick review, I updated the PR to cover all the API and fix the style inconsistency.

**Comment by Sija** [comment]
That's a serious bloat within the docs, whereas it pretty easy to just check main module doc comment for a needed `require`, c'mon...

**Comment by bcardiff** [comment]
@Sija having self-contained examples is good. Even at the cost of 2 duplicate lines IMO.
Maybe for libs where more context is needed that does not apply. But at least for the stdlib is something I think we should aim.

**Comment by straight-shoota** [comment]
I've been questioning the noise this generates, but having them is better than not having them.

And it could perhaps in the future be an option to optimize the display of API docs. It's easy to recognize repeating `require` statements and make them fade or disappear or whatever. It doesn't work the other way around.

**Comment by bcardiff** [comment]
@Maroo-b I think we are ready to merge this as is.
The issue you notice is fixed in #7569 
I think there is no conflict between both PR so it's better to merge this, then the other and then if wanted the [M-Z] range can be tackled.

ok?

**Comment by Maroo-b** [comment]
@bcardiff I guess it can be merged, also I already covered [A-Z] (which means every doc page for **0.27.2** version.

**Comment by Maroo-b** [comment]
@bcardiff @straight-shoota I would like to contribute more to Crystal in my free time, is there a tag or a board for highest priority issues or tasks that I can work on?

**Comment by straight-shoota** [comment]
Thank you very much @Maroo-b! I'm looking forward for additional contributions from you.

There is no real priority list. Important issues are usually ones [tagged with the next milestone](https://github.com/crystal-lang/crystal/issues?utf8=%E2%9C%93&q=is%3Aopen+milestone%3A0.28.0+is%3Aissue) but they're often more complex bugs (or features that are already worked). Another starting point are the [`community:` labels](https://github.com/crystal-lang/crystal/labels), for example [`community:to-implement`](https://github.com/crystal-lang/crystal/labels/community%3Ato-implement). Apart from that, just pick whatever you're interested in. You can browse the list of open issues and take what you like ;)

**Comment by Maroo-b** [comment]
@straight-shoota thank you for your quick feedback!

Sure, I'll check the labeled issues :)

**Pull request merged**